encrypter.cpp

```diff
@@ -0,0 +1,202 @@
+#include <iostream>
+#include <string.h>
+#include <fstream>
+#include <ios>
+#include <random>
+#include <sstream>
+
+
+//=====put_error=====//
+void put_error(const std::string& error_message)
+{
+  std::cout << error_message;
+  exit(1);
+}
+
+//=====shuffle=====//
+
+void shuffle(std::vector<char>& input)
+{
+  std::default_random_engine generator;
+  std::uniform_int_distribution<int> distribution(0, input.size() - 1);
+  int i = 0;
+
+  for (std::vector<char>::iterator iter = input.begin(); iter < input.end(); ++i, ++iter)
+  {
+    int random = distribution(generator);
+    char tmp = input[i];
+    input[i] = input[random];
+    input[random] = tmp;
+  } 
+}
+
+//=====replace_extension=====//
+std::string replace_extension(const std::string& filename, const std::string& new_extension)
+{
+  int i;
+  std::string new_name;
+  std::string::const_iterator iter = new_extension.begin();
+
+  for (i = 0; filename[i] != '.'; ++i);
+
+  ++i;
+  if (filename[i] == '.')
+    put_error("Oh, snap! Something went wrong.\n");
+
+  new_name = filename.substr(0, i + 1);
+
+  for (int k = 0; iter < new_extension.end(); ++k, ++i, ++iter)
+    new_name[i] = new_extension[k];
+
+  return new_name;
+}
+
+//=====read_file=====//
+std::vector<char> read_file(const std::string& filename, const std::string error_message, std::streampos& size)
```

Comment:
You return vector without moving semantic. And vector doesn't have embedded moving semantic (string has). So you going to unnecessarily copy all the data in this vector.